winrm/client.go

```diff
@@ -23,21 +29,38 @@ type Client struct {
 
 // NewClient will create a new remote client on url, connecting with user and password
 // This function doesn't connect (connection happens only when CreateShell is called)
-func NewClient(endpoint *Endpoint, user, password string) (client *Client, err error) {
+func NewClient(endpoint *Endpoint, user, password string, auth AuthType) (client *Client, err error) {
```

Comment:
Please don't add a parameter to this function, its API should be stable.
I think we don't need to know if we should use CertAuth or BasicAuth, we can infer this from the given parameters. If the user sent in a client certificate, then we should use it, if the client sent in a username and password then we should use them.
And BTW, it's also possible to use both CertAuth and BasicAuth at the same time since they aren't at the same level.
I would suggest removing this parameter and infer from the Endpoint values.